src/ui/Frame.cpp

```diff
@@ -0,0 +1,1164 @@
+	Field* field = new Field(*this, len);
+	field->setName(name);
+	return field;
+}
+
+Frame::image_t* Frame::addImage(const SDL_Rect& pos, const Uint32& color, const char* image, const char* name) {
+	if (!image || !name) {
+		return nullptr;
+	}
+	image_t* imageObj = new image_t();
+	imageObj->pos = pos;
+	imageObj->color = color;
+	imageObj->name = name;
+	imageObj->path = image;
+	images.push_back(imageObj);
+	return imageObj;
+}
+
+Slider* Frame::addSlider(const char* name) {
+	if (!name) {
+		return nullptr;
+	}
+	Slider* slider = new Slider(*this);
+	slider->setName(name);
+	sliders.push_back(slider);
+	return slider;
+}
+
+Frame::entry_t* Frame::addEntry(const char* name, bool resizeFrame) {
+	entry_t* entry = new entry_t();
+	entry->name = name;
+	entry->color = 0xffffffff;
+	entry->image = nullptr;
+	list.push_back(entry);
+
+	if (resizeFrame) {
+		resizeForEntries();
+	}
+
+	return entry;
+}
+
+void Frame::removeSelf() {
+	toBeDeleted = true;
+}
+
+void Frame::clear() {
+	// delete frames
+	while (frames.size()) {
+		delete frames.front();
+		frames.erase(frames.begin());
+	}
+
+	// delete buttons
+	while (buttons.size()) {
+		delete buttons.front();
+		buttons.erase(buttons.begin());
+	}
+
+	// delete fields
+	while (fields.size()) {
+		delete fields.front();
+		fields.erase(fields.begin());
+	}
+
+	// delete images
+	while (images.size()) {
+		delete images.front();
+		images.erase(images.begin());
+	}
+
+	// delete sliders
+	while (sliders.size()) {
+		delete sliders.front();
+		sliders.erase(sliders.begin());
+	}
+
+	// delete list
+	while (list.size()) {
+		delete list.front();
+		list.erase(list.begin());
+	}
+	selection = list.end();
+}
+
+void Frame::clearEntries() {
+	while (list.size()) {
+		delete list.front();
+		list.erase(list.begin());
+	}
+	selection = list.end();
+}
+
+bool Frame::remove(const char* name) {
+	for (auto it = frames.begin(); it != frames.end(); ++it) {
+		Frame* frame = *it;
+		if (strcmp(frame->getName(), name) == 0) {
+			delete frame;
+			frames.erase(it);
+			return true;
+		}
+	}
+	for (auto it = buttons.begin(); it != buttons.end(); ++it) {
+		Button* button = *it;
+		if (strcmp(button->getName(), name) == 0) {
+			delete button;
+			buttons.erase(it);
+			return true;
+		}
+	}
+	for (auto it = fields.begin(); it != fields.end(); ++it) {
+		Field* field = *it;
+		if (strcmp(field->getName(), name) == 0) {
+			delete field;
+			fields.erase(it);
+			return true;
+		}
+	}
+	for (auto it = images.begin(); it != images.end(); ++it) {
+		image_t* image = *it;
+		if (strcmp(image->name.c_str(), name) == 0) {
+			delete image;
+			images.erase(it);
+			return true;
+		}
+	}
+	for (auto it = sliders.begin(); it != sliders.end(); ++it) {
+		Slider* slider = *it;
+		if (strcmp(slider->getName(), name) == 0) {
+			delete slider;
+			sliders.erase(it);
+			return true;
+		}
+	}
+	return false;
+}
+
+bool Frame::removeEntry(const char* name, bool resizeFrame) {
+	entry_t* prevEntry = nullptr;
```

Comment:
What is prevEntry used for?

Reply:
Left that one in by mistake... I'll remove it